packages/templates/nextjs-marketing-demo/src/components/ButtonComponentRegistration.tsx

```diff
@@ -0,0 +1,126 @@
+import { Button } from 'antd';
+import Icon from './Icon';
+
+type ButtonComponentProps = Omit<React.ComponentProps<typeof Button>, 'children' | 'icon'> & {
+  text: string;
+  icon?: React.ComponentProps<typeof Icon>['icon'];
+};
+
+export const ButtonComponentRegistration = {
+  component: (props: ButtonComponentProps) => {
+    const { color, icon, variant, text, ...rest } = props;
+    return (
+      <>
+        <Button
+          color={color ? color : undefined}
+          variant={variant ? variant : undefined}
```

Comment:
oops, meant to check if it was undefined or an empty string. Since the UI doesn't support undefined as custom design values, it sends back an empty string, which doesn't pass the components type checking.